--- a/contrib/slapd-modules/passwd/argon2/Makefile	2021-06-03 18:40:31.000000000 +0000
+++ b/contrib/slapd-modules/passwd/argon2/Makefile	2023-02-23 16:30:20.623741751 +0000
@@ -2,7 +2,7 @@
 
 LDAP_SRC = ../../../..
 LDAP_BUILD = ../../../..
-LDAP_INC = -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
+LDAP_INC = -I/opt/zextras/common/include -I$(LDAP_BUILD)/include -I$(LDAP_SRC)/include -I$(LDAP_SRC)/servers/slapd
 LDAP_LIB = $(LDAP_BUILD)/libraries/libldap_r/libldap_r.la \
 	$(LDAP_BUILD)/libraries/liblber/liblber.la
 
