diff '--color=auto' -Naur a/memcached.c b/memcached.c
--- a/memcached.c	2026-05-21 12:59:53.421431751 +0200
+++ b/memcached.c	2026-05-21 12:59:59.771552489 +0200
@@ -47,6 +47,10 @@
 #include <sysexits.h>
 #include <stddef.h>
 
+#ifdef HAVE_SYSTEMD
+#include <systemd/sd-daemon.h>
+#endif
+
 #ifdef HAVE_GETOPT_LONG
 #include <getopt.h>
 #endif
@@ -6180,6 +6184,11 @@
     /* Initialize the uriencode lookup table. */
     uriencode_init();
 
+#ifdef HAVE_SYSTEMD
+    sd_notify(0, "READY=1\n"
+                  "STATUS=memcached is ready, accepting connections");
+#endif
+
     /* enter the event loop */
     while (!stop_main_loop) {
         if (event_base_loop(main_base, EVLOOP_ONCE) != 0) {
