diff '--color=auto' -Naur a/Makefile.PL b/Makefile.PL
--- a/Makefile.PL	2025-06-30 17:00:52.087453804 +0200
+++ b/Makefile.PL	2025-06-30 17:04:28.169794344 +0200
@@ -4,10 +4,7 @@
 
 use 5.006;
 use ExtUtils::MakeMaker 6.48;
-use Crypt::OpenSSL::Guess qw(openssl_inc_paths openssl_lib_paths openssl_version);
 
-my ($major, $minor, $patch) = openssl_version(); 
-print "OpenSSL version: $major.$minor $patch", "\n";
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
 
@@ -16,6 +13,10 @@
   $libs = $libs . ' -lz';
 }
 
+my $KERBEROS_INCLUDE = "/usr/kerberos/include";
+
+my $include = -d $KERBEROS_INCLUDE ? "-I${KERBEROS_INCLUDE}" : "";
+
 WriteMakefile(
     'NAME'             => 'Crypt::OpenSSL::RSA',
     'AUTHOR'           => 'Ian Robertson <iroberts@cpan.org>',
@@ -30,12 +31,11 @@
         'Test::More'             => 0,
     },
     'OBJECT' => 'RSA.o',
-    'LIBS'   => [openssl_lib_paths() . $libs],
-    'LDDLFLAGS' => openssl_lib_paths() . ' ' . $Config{lddlflags},
+    'LIBS'    => ($^O eq 'MSWin32') ? ['-lssl32 -leay32'] : ['-lssl -lcrypto'],
     'DEFINE' => '-DPERL5 -DOPENSSL_NO_KRB5',
 
     # perl-5.8/gcc-3.2 needs -DPERL5, and redhat9 likes -DOPENSSL_NO_KRB5
-    'INC' => openssl_inc_paths(),    # e.g., '-I/usr/include/other'
+    'INC'	=> $include,     # e.g., '-I/usr/include/other'
 
     'dist'  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
     'clean' => { FILES    => 'Crypt-OpenSSL-RSA-*' },
@@ -43,9 +43,6 @@
         recommends => {
             'Crypt::OpenSSL::Bignum' => 0,
         },
-        configure_requires => {
-            'Crypt::OpenSSL::Guess' => '0.11',
-        },
         build_requires => {
             'Test' => 0,    # For testing
         },
