diff '--color=auto' -Naur a/mime_alter.c b/mime_alter.c
--- a/mime_alter.c	2008-11-16 09:45:45.000000000 +0100
+++ b/mime_alter.c	2022-01-17 14:55:09.152489040 +0100
@@ -1369,10 +1369,16 @@
 	if (0 == stat(fname, &st))
 	{
 		char *p;
+		
+		if (0 == st.st_size) {
+			DAM LOGGER_log("%s:%d:AM_disclaimer_load_text:DEBUG: Ignoring empty file '%s'", FL, fname);
+			return -1;
+		}
+
 		p = malloc((st.st_size +1)*sizeof(char));  // We have to add 1 so that we delimit the data with a \0
 		if (p == NULL) {
 			LOGGER_log("%s:%d:AM_disclaimer_load_text:ERROR: Unable to allocate memory to load file '%s'", FL, fname);
-			return 1;
+			return -1;
 		}
 
 		if (p)
@@ -1399,6 +1405,7 @@
 
 	} else {
 		LOGGER_log("%s:%d:AM_disclaimer_load_text:ERROR: Cannot stat '%s' (%s)",FL,fname,strerror(errno));
+		return -1;
 	}
 
 
@@ -1726,7 +1733,7 @@
 	{
 		if (glb.disclaimer_plain_type == AM_DISCLAIMER_TYPE_FILENAME)
 		{
-			AM_disclaimer_load_text( glb.disclaimer_plain, &(dd->disclaimer_text_plain) );
+			dud_text = AM_disclaimer_load_text( glb.disclaimer_plain, &(dd->disclaimer_text_plain) );
 		}
 		else
 		{
@@ -1742,7 +1749,7 @@
 	{
 		if (glb.disclaimer_HTML_type == AM_DISCLAIMER_TYPE_FILENAME)
 		{
-			AM_disclaimer_load_text( glb.disclaimer_HTML, &(dd->disclaimer_text_HTML) );
+			dud_html = AM_disclaimer_load_text( glb.disclaimer_HTML, &(dd->disclaimer_text_HTML) );
 		}
 		else
 		{
@@ -1757,7 +1764,7 @@
 	{
 		if (glb.disclaimer_b64_type == AM_DISCLAIMER_TYPE_FILENAME)
 		{
-			AM_base64_decode_to_buffer( glb.disclaimer_b64, &(dd->disclaimer_text_b64) );
+			dud_b64 = AM_base64_decode_to_buffer( glb.disclaimer_b64, &(dd->disclaimer_text_b64) );
 		} else {
 			dd->disclaimer_text_b64 = strdup( glb.disclaimer_b64 );
 			AM_base64_decode_buffer( dd->disclaimer_text_b64, strlen(dd->disclaimer_text_b64));
@@ -1767,9 +1774,9 @@
 	}
 
 	// If our disclaiemrs are all 'dud's, then we should just return with an error.
-	if ((dud_html == 1)&&(dud_text == 1)&&(dud_b64 == 1))
+	if ((dud_html != 0)&&(dud_text != 0)&&(dud_b64 != 0))
 	{
-		LOGGER_log("%s:%d:AM_load_disclaimers: Neither the Plain-text , HTML or BASE64 disclaimer were valid to insert, skipping disclaimer-insertion routine\n");
+		LOGGER_log("%s:%d:AM_load_disclaimers: Neither the Plain-text, HTML or BASE64 disclaimer were valid to insert, skipping disclaimer-insertion routine\n", FL);
 		return -1;
 	}
 
